cmd/gateway/s3x/gateway-s3x-multipart.go

```diff
@@ -107,17 +107,24 @@ func (x *xObjects) ListObjectParts(
 		MaxParts:         maxParts,
 		PartNumberMarker: partNumberMarker,
 	}
-	parts, err := x.ledgerStore.GetObjectParts(uploadID)
+	m, unlock, err := x.ledgerStore.GetObjectDetails(uploadID)
+	defer unlock()
 	if err != nil {
 		return lpi, x.toMinioErr(err, bucket, object, uploadID)
 	}
-	for _, part := range parts {
+	if m.GetObjectInfo().GetBucket() != bucket ||
+		m.GetObjectInfo().GetName() != object {
+		return lpi, x.toMinioErr(ErrInvalidUploadID, bucket, object, uploadID)
+	}
+
+	for _, part := range m.ObjectParts {
 		lpi.Parts = append(lpi.Parts, minio.PartInfo{
 			PartNumber: int(part.GetNumber()),
-			ETag:       minio.ToS3ETag(part.GetEtag()),
+			ETag:       minio.ToS3ETag(part.GetDataHash()),
```

Comment:
I think this is a typo? It should be `part.GetETag` not `part.GetDataHash`

Reply:
I removed the Etag field and started using DataHash as the Etag

Reply:
makes sense